auto-complete.js

```diff
@@ -117,50 +126,69 @@ var autoComplete = (function(){
             }
             that.parentElement.appendChild(that.sc);
 
-            live('autocomplete-suggestion', 'mouseleave', function(e){
+            live('autocomplete-suggestion', 'mouseleave', function (e) {
                 var sel = that.sc.querySelector('.autocomplete-suggestion.selected');
-                if (sel) setTimeout(function(){ sel.className = sel.className.replace('selected', ''); }, 20);
+                if (sel) setTimeout(function () { sel.className = sel.className.replace('selected', ''); }, 20);
             }, that.sc);
 
-            live('autocomplete-suggestion', 'mouseover', function(e){
+            live('autocomplete-suggestion', 'mouseover', function (e) {
                 var sel = that.sc.querySelector('.autocomplete-suggestion.selected');
                 if (sel) sel.className = sel.className.replace('selected', '');
                 this.className += ' selected';
             }, that.sc);
 
-            live('autocomplete-suggestion', 'mousedown', function(e){
+            live('autocomplete-suggestion', 'mousedown', function (e) {
                 if (hasClass(this, 'autocomplete-suggestion')) { // else outside click
                     var v = this.getAttribute('data-val');
+                    if (o.queryHistory) {
+                        if (e.path[1].className === 'local-suggestion-remove-button') {
```

Comment:
What is this `e.path[1]`?
Also, you shouldn't check if the `className` is equal to some value, it's fragile and will easily break.
Use [classList.contains method](https://developer.mozilla.org/en-US/docs/Web/API/element/classList)

Reply:
I used the path to identify if the suggestion or the remove button was clicked. 
Already changed to a better version with the `classList.contains` method you suggested.